pkg/skaffold/build/cluster/sources/sources.go

```diff
@@ -82,28 +82,29 @@ func podTemplate(clusterDetails *latest.ClusterDetails, artifact *latest.KanikoA
 					Args:            args,
 					ImagePullPolicy: v1.PullIfNotPresent,
 					Env:             env,
-					VolumeMounts: []v1.VolumeMount{
-						{
-							Name:      constants.DefaultKanikoSecretName,
-							MountPath: "/secret",
-						},
-					},
-					Resources: resourceRequirements(clusterDetails.Resources),
+					Resources:       resourceRequirements(clusterDetails.Resources),
 				},
 			},
 			RestartPolicy: v1.RestartPolicyNever,
-			Volumes: []v1.Volume{{
-				Name: constants.DefaultKanikoSecretName,
-				VolumeSource: v1.VolumeSource{
-					Secret: &v1.SecretVolumeSource{
-						SecretName: clusterDetails.PullSecretName,
-					},
-				},
-			},
-			},
 		},
 	}
+	if clusterDetails.PullSecretName != "" {
```

Comment:
this function is becoming really long, can we refactor this into 

```suggestion+15
	if clusterDetails.PullSecretName != "" {
	  vms, vs := getVolumeMountsAndVolumes()
	  pod.Spec.Containers[0].VolumeMounts = vms
	  pod.Spec.Volumes = vs  
```

Reply:
ya sure